src/main/java/sdu/sem2/se17/persistence/data/CreditHandlerImpl.java

```diff
@@ -4,43 +4,163 @@
 import sdu.sem2.se17.domain.persistenceinterface.CreditHandler;
 import sdu.sem2.se17.persistence.db.DataSource;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.Optional;
 
 public class CreditHandlerImpl implements CreditHandler {
     private final DataSource dataSource;
+    private ParticipantHandlerImpl participantHandler;
 
-    public CreditHandlerImpl(DataSource dataSource) {
+    public CreditHandlerImpl(DataSource dataSource, ParticipantHandlerImpl participantHandler) {
         this.dataSource = dataSource;
+        this.participantHandler = participantHandler;
     }
 
     @Override
     public ArrayList<Credit> findByProductionId(long id) {
-        return null;
-    }
+        ArrayList<Credit> result = new ArrayList<>();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                    SELECT * FROM Credit
+                    WHERE productionId = ?
+                    RETURNING *
```

Comment:
I believe the **RETURNING** statement should be removed, as the **SELECT** query by default returns the values.
```suggestion
```